common/app/model/liveblog/BlockElement.scala

```diff
@@ -1,24 +1,52 @@
 package model.liveblog
 
 import com.gu.contentapi.client.model.v1.ElementType.{Map => _, _}
-import com.gu.contentapi.client.model.v1.{SponsorshipType, BlockElement => ApiBlockElement, Sponsorship => ApiSponsorship}
+import com.gu.contentapi.client.model.v1.{ElementType, SponsorshipType, BlockElement => ApiBlockElement, Sponsorship => ApiSponsorship}
 import model.{AudioAsset, ImageAsset, ImageMedia, VideoAsset}
 import play.api.libs.json._
 
 sealed trait BlockElement
 case class TextBlockElement(html: Option[String]) extends BlockElement
+case class TweetBlockElement(html: Option[String]) extends BlockElement
+case class PullquoteBlockElement(html: Option[String]) extends BlockElement
 case class ImageBlockElement(media: ImageMedia, data: Map[String, String], displayCredit: Option[Boolean]) extends BlockElement
 case class AudioBlockElement(assets: Seq[AudioAsset]) extends BlockElement
 case class GuVideoBlockElement(assets: Seq[VideoAsset], imageMedia: ImageMedia, data: Map[String, String]) extends BlockElement
 case class VideoBlockElement(data: Map[String, String]) extends BlockElement
 case class EmbedBlockElement(html: Option[String], safe: Option[Boolean], alt: Option[String]) extends BlockElement
 case class ContentAtomBlockElement(atomId: String) extends BlockElement
+case class InteractiveBlockElement(html: Option[String]) extends BlockElement
+case class CommentBlockElement(html: Option[String]) extends BlockElement
+case class TableBlockElement(html: Option[String]) extends BlockElement
+case class WitnessBlockElement(html: Option[String]) extends BlockElement
+case class DocumentBlockElement(html: Option[String]) extends BlockElement
+case class InstagramBlockElement(html: Option[String]) extends BlockElement
+case class VineBlockElement(html: Option[String]) extends BlockElement
+case class MapBlockElement(html: Option[String]) extends BlockElement
+case class UnknownBlockElement(html: Option[String]) extends BlockElement
+
+case class MembershipBlockElement(
```

Comment:
Again, I'm guessing this is just an external model we've copied in, but why is everything optional here?

Reply:
Everything is probably optional because the fields are optional on the thrift models themselves